File: app/src/main/java/com/android/retrofitsampleapp2/iu/GitUsersViewHolder.java
package com.android.retrofitsampleapp2.iu;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.android.retrofitsampleapp2.R;
import com.android.retrofitsampleapp2.domain.GitUserEntity;

public class GitUsersViewHolder extends RecyclerView.ViewHolder {

    private final TextView titleTextView = itemView.findViewById(R.id.title_text_view);
    private final TextView subtitleTextView = itemView.findViewById(R.id.subtitle_text_view);

    public GitUsersViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    public void bind(GitUserEntity gitUserEntity) {
        titleTextView.setText(gitUserEntity.getLogin());
        subtitleTextView.setText(gitUserEntity.getNodeId());
    }
}

File: app/src/main/java/com/android/retrofitsampleapp2/iu/users/GitUsersViewHolder.kt
package com.android.retrofitsampleapp2.iu.users

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.android.retrofitsampleapp2.R
import com.android.retrofitsampleapp2.domain.GitUserEntity

class GitUsersViewHolder(itemView: View, listener: GitUsersAdapter.OnItemClickListener) :
    RecyclerView.ViewHolder(itemView) {
    private val titleUserTextView = itemView.findViewById<TextView>(R.id.title_user_text_view)
    private val subtitleUserTextView = itemView.findViewById<TextView>(R.id.subtitle_user_text_view)
    private lateinit var user: GitUserEntity

    fun bind(gitUserEntity: GitUserEntity) {
        user = gitUserEntity //сохранили user в текущей карточке
        titleUserTextView.text = gitUserEntity.login
        subtitleUserTextView.text = gitUserEntity.nodeId
    }

    init {
        // обработка нажатия
        // при нажатии на элемент происходит вызов listener и передается в него user который положили в bind
        itemView.setOnClickListener { _ ->
            user.let {
                listener.onItemClick(
                    it
                )
            }
        }
    }
}